query/src/main/java/org/infinispan/query/clustered/ClusteredQueryInvoker.java

```diff
@@ -69,6 +68,9 @@ List<QueryResponse> broadcast(ClusteredQueryOperation operation) {
                                                rpcManager.getSyncRpcOptions()).toCompletableFuture();
             }).map(a -> a.thenApply(r -> (QueryResponse) r.getResponseValue())).collect(Collectors.toList());
 
+      // then, invoke on own node
+      CompletionStage<QueryResponse> localResponse = localInvoke(localCommand);
```

Comment:
@fax4ever I just want to bring awareness to this change. I was doing some JFR and the computation is not offloaded to a different thread.

Reply:
So you move it after the calls the asynchronous call. Great.